base/dates/query.jl

```diff
@@ -20,49 +17,55 @@ const MONTHDAYS = [0,31,59,90,120,151,181,212,243,273,304,334]
 dayofyear(y,m,d) = MONTHDAYS[m] + d + (m > 2 && isleapyear(y))
 
 ### Days of the Week
-"""
-    dayofweek(dt::TimeType) -> Int64
+@enum DayOfWeek Monday=1 Tuesday=2 Wednesday=3 Thursday=4 Friday=5 Saturday=6 Sunday=7
 
-Returns the day of the week as an `Int64` with `1 = Monday, 2 = Tuesday, etc.`.
 """
-dayofweek(dt::TimeType) = dayofweek(days(dt))
+    DayOfWeek(n::Integer)
 
-const Monday,Tuesday,Wednesday,Thursday,Friday,Saturday,Sunday = 1,2,3,4,5,6,7
-const Mon,Tue,Wed,Thu,Fri,Sat,Sun = 1,2,3,4,5,6,7
-const english_daysofweek = Dict(1=>"Monday",2=>"Tuesday",3=>"Wednesday",
-                                4=>"Thursday",5=>"Friday",6=>"Saturday",7=>"Sunday")
-const VALUETODAYOFWEEK = Dict{String,Dict{Int,String}}("english"=>english_daysofweek)
-const english_daysofweekabbr = Dict(1=>"Mon",2=>"Tue",3=>"Wed",
-                                    4=>"Thu",5=>"Fri",6=>"Sat",7=>"Sun")
-const VALUETODAYOFWEEKABBR = Dict{String,Dict{Int,String}}("english"=>english_daysofweekabbr)
-dayname(dt::Integer;locale::AbstractString="english") = VALUETODAYOFWEEK[locale][dt]
+An `Enum` with `DayOfWeek(1) == Monday`, `DayOfWeek(2) == Tuesday`, etc. Accepts an
+arbitrary-sized integer, from which days will be computed modulo 7.
 
+```jldoctest
+julia> DayOfWeek(8) == Monday
+true
+```
 """
-    dayabbr(dt::TimeType; locale="english") -> AbstractString
+DayOfWeek(n::Integer) = convert(DayOfWeek, mod1(n, 7))
 
-Return the abbreviated name corresponding to the day of the week of the `Date` or `DateTime`
-in the given `locale`.
 """
-dayabbr(dt::Integer;locale::AbstractString="english") = VALUETODAYOFWEEKABBR[locale][dt]
+    DayOfWeek(dt::TimeType)
```

Comment:
I think I'd like to still keep this lowercase `dayofweek`, it follows the more general pattern in Dates with types being proper-cased and accessors being lowercased.